README.md

```diff
@@ -1,2 +1,59 @@
-# puppetlabs-ruby_task_helper
-A ruby library for running tasks
+# Ruby Task Helper
+
+This is a helper library for writing tasks. It provides a class that handles error generation, simplifies JSON input and output, and makes testing your task easier.
+
+#### Table of Contents
+
+1. [Description](#description)
+2. [Setup](#setup)
+3. [Usage](#usage)
+
+## Description
+
+## Setup
+
+Include this module in a [Puppetfile]():
+
+```
+mod puppetlabs-ruby_task_helper
+```
+
+## Usage
+
+When writing your task include the library in your script, extend the `TaskClass` module, and write the `task()` function. The `task()` function **must** take a hash of parameters as input (even if it's an empty hash), and **must** return a hash. The following is an example of a task that uses the library
+
+`mymodule/tasks/task.rb`
+```
+require_relative "#{ENV['_installdir']}/ruby_task_helper.rb"
+
+include TaskClass
+extend TaskClass
```

Comment:
this should probably be it's own class. otherwise tasks that require each other may behave poorly